File: Data Structures & Algorithms/duplicate-integer/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     */
    hasDuplicate(nums) {
        const set = new Set();

  let contains = false;

  function iterate(nextNums) {
    if (nextNums.length === 0) {
      return;
    }

    const num = nextNums[0];

    if (set.has(num)) {
      contains = true;
    } else {
      set.add(num);
    }

    iterate(nextNums.slice(1));
  }

  iterate(nums);

  return contains
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        # [-1,0,1,2,3]
        # [0, ]
        ans = []

        for i in range(len(nums)):
            r, l = i + 1, 0
            s = 1

            while l < i:
                s = s * nums[l]
                l = l + 1

            while r < len(nums):
                s = s * nums[r]
                r = r + 1
            
            ans.append(s)
        
        return ans